client/src/components/NewListDialog.jsx

```diff
@@ -0,0 +1,100 @@
+import {
+  makeStyles,
+  Grid,
+  Typography,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  Button,
+  TextField,
+} from "@material-ui/core";
+import { useState } from "react";
+import axios from "axios";
+
+const useStyles = makeStyles(() => ({
+  button: {
+    height: 56,
+    borderRadius: 28,
+    width: 142,
+  },
+}));
+
+const NewListDialog = ({ isOpen, onClose }) => {
+  const classes = useStyles();
+
+  const [title, setTitle] = useState("");
+  const [coverImageUrl, setCoverImageUrl] = useState(
+    "http://example.com/image"
+  );
+  const [awaitingResponse, setAwaitingResponse] = useState(false);
+
+  const handleCoverImageUrlChange = (e) => {
+    setCoverImageUrl(e.target.value);
+  };
+  const handleTitleChange = (e) => {
+    setTitle(e.target.value);
+  };
+  //TODO: Should be in useEffect
+  const handleSubmit = async () => {
+    try {
+      setAwaitingResponse(true);
+      //TODO: Use environment variable for baseUrl
+      await axios
+        .create({ baseUrl: "localhost:3001", withCredentials: true })
```

Comment:
you shouldn't need to define the baseUrl since we're using a proxy, we can just make requests to `/lists`